src/vsl/parser/grammar/assignment.ne

```diff
@@ -2,16 +2,21 @@
 @include "primitives.ne"
 
 # Parses assignment statements
-AssignmentStatement -> AssignmentType _ TypedIdentifier ( _ "=" _ Expression {% nth(3) %}):? {%
-    (d, l) => new t.AssignmentStatement(
-        d[0] === "let" ? t.AssignmentType.Constant : t.AssignmentType.Variable,
-        d[2],
-        d[3],
-        l
-    )
-%}
+AssignmentStatement
+   -> AssignmentType _ TypedIdentifier ( _ "=" _ Expression {% nth(3) %}):? {%
+        (data, location) => {
+            var type;
+            if (data[0] === "let") {
```

Comment:
Since this is JS I think a lookup-object would be most idiomatic:
```javascript
{
    "let": t.AssignmentType.Constant
    // if we wanted to add more we could just do:
    "ref": t.AssignmentType.Reference
}[data[0]] || t.AssignmentType.Variable
```
as we are likely going to expand the amount of "AssignmentTypes". The reason I'm saying this is that when we expand it an `if..else if..else` is going to be over-verbose for expressing a simple alias. You could make that change or if you don't want to you could just merge it and we'll figure it out later